homeassistant/bootstrap.py

```diff
@@ -150,28 +158,32 @@ def from_config_dict(config: Dict[str, Any],
     _LOGGER.info("Home Assistant core initialized")
 
     # stage 0, load logging components
-    for component in components:
-        if component in LOGGING_COMPONENT:
-            hass.async_create_task(
-                async_setup_component(hass, component, config))
-
-    await hass.async_block_till_done()
+    await asyncio.gather(*[
+        async_setup_component(hass, component, config)
+        for component in components
+        if component in LOGGING_COMPONENT
+    ])
+    # I have audited the logging components they don't create any
+    # critical tasks so we don't need to block_till_done here.
 
     # stage 1
-    for component in components:
-        if component in FIRST_INIT_COMPONENT:
-            hass.async_create_task(
-                async_setup_component(hass, component, config))
+    asyncio.gather(*[
```

Comment:
await

Reply:
Doesn't need to, the line below awaits them all.